test/runtime.spec.ts

```diff
@@ -2,14 +2,60 @@ import * as jsc from 'jsverify';
 import * as _ from 'lodash';
 import * as assert from 'assert';
 import { promisify } from 'util';
-import { make, pmap, set, map, getAll } from '../src/runtime';
+import * as server from '../src/server';
+import { make, pmap, set, map, getAll, json } from '../src/runtime';
 import { TestClass } from './test-class';
 
 const getWithTraversalPath = (dict: any, path: string[]): any => {
   return path.reduce((toBeTraversed: any, nextHop: string) => {
     return toBeTraversed[nextHop];
   }, dict);
 };
+describe('responses', () => {
+  it('returns headers with correct type', () => {
+    const res = json(100, null, { value: 1 });
+
+    // @ts-expect-error
+    res.headers.wrongValue;
+    expect(res.headers.value).toEqual(1);
+  });
+
+  it('defaults headers to record type if none are given', () => {
+    const res = json(100, null);
+
+    res.headers.anyValue;
+    res.headers.value;
+    expect(res.headers).toEqual({});
+  });
+
+  it('does not require headers when any record is accepted', () => {
+    json<number, null, Record<string, any>>(100, null);
+  });
+
+  it('infers empty headers', () => {
+    json(100, null);
+  });
+
+  function hasType<T>(t: T) {}
+
+  it('requires headers when those are mandatory', () => {
+    // @ts-expect-error
+    hasType<server.Response<100, 'application/json', null, { value: number }>>(
```

Comment:
this fails now. which is a bit not nice.

Reply:
I changed it to having a function to setting the headers of a response. Otherwise fixing this would require some complicated type logic that I don't think is worth it